packages/react/src/components/ActionList/ActionList.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {
+  type MutableRefObject,
+  forwardRef,
+  useCallback,
+  useRef
+} from 'react';
+import classnames from 'classnames';
+import type { ListboxOption } from '../Listbox/ListboxContext';
+import Listbox from '../Listbox';
+import {
+  type ActionListSelectionType,
+  type onActionCallbackFunction,
+  type onActionEvent,
+  ActionListProvider
+} from './ActionListContext';
+import { useActionListContext } from './ActionListContext';
+
+interface ActionListProps extends React.HTMLAttributes<HTMLUListElement> {
+  children: React.ReactNode;
+
+  /** Limits the amount of selections that can be made within an action list */
+  selectionType?: ActionListSelectionType | null;
+
+  /** A callback function that is called when an action list item is selected. */
+  onAction?: onActionCallbackFunction;
+}
+
+const ActionList = forwardRef<HTMLUListElement, ActionListProps>(
+  ({ selectionType = null, onAction, className, children, ...props }, ref) => {
+    const actionListContext = useActionListContext();
+    const activeElement = useRef<
+      HTMLLIElement | HTMLAnchorElement
+    >() as MutableRefObject<HTMLLIElement | HTMLAnchorElement>;
+
+    const handleActiveChange = useCallback((value: ListboxOption) => {
+      activeElement.current = value?.element as
+        | HTMLLIElement
+        | HTMLAnchorElement;
+    }, []);
+
+    const handleAction = useCallback(
+      (key: string, event: onActionEvent) => {
+        if (typeof onAction === 'function') {
+          onAction(key, event);
+        }
+      },
+      [onAction]
+    );
+
+    const handleKeyDown = useCallback(
+      (event: React.KeyboardEvent<HTMLUListElement | HTMLAnchorElement>) => {
+        if (event.defaultPrevented) {
+          return;
+        }
+
+        // Since focus is managed in the action list using `aria-activedescendant`
+        // we want to simulate a keypress on the current active list item
+        if (event.key === 'Enter' || event.key === ' ') {
+          activeElement.current?.click();
+        }
+      },
+      []
+    );
+
+    return (
+      // Note: we should be able to use listbox without passing a prop
+      // value for "multiselect"
+      // see: https://github.com/dequelabs/cauldron/issues/1890
+      // @ts-expect-error this should be allowed
+      <Listbox
+        ref={ref}
+        /* Listbox comes with an explicit role of "listbox", but we want to either
+         * use the role from props, or default to the intrinsic role */
+        // eslint-disable-next-line jsx-a11y/aria-role
+        role={undefined}
+        // aria-multiselectable is valid for listbox roles, but not list or menu roles
+        aria-multiselectable={
+          actionListContext.role === 'listbox' ? undefined : null
+        }
```

Comment:
We aren't setting the value of [aria-multiselectable](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-multiselectable) here. Isn't this the same as omitting the property entirely?

Reply:
Yes, that's on purpose. It's to override the value of `aria-multiselectable` here: https://github.com/dequelabs/cauldron/blob/1266d64e7d0ea8dca2af3b00e1a6b20dc18b90f4/packages/react/src/components/Listbox/Listbox.tsx#L287

When spread, `undefined` won't override the default value from listbox, but `null` will override the value since this is not a valid property when in a non-listbox role.

Reply:
Could we add a comment it's not obvious from the code that that's the reason why?